domain/modelconfig/service/service.go

```diff
@@ -351,6 +361,16 @@ func (s *Service) updateModelConfigValidator(
 	return agg
 }
 
+// GetModelSecretBackend returns the secret backend name for the given model UUID.
+func (s *Service) GetModelSecretBackend(ctx context.Context, modelUUID coremodel.UUID) (string, error) {
```

Comment:
I don't think either of these functions should be living on model config. We are just confusing domains and business logic.

You should have and be using a secret backend service. Because the secret backend service is on the controller database and you need to know the model  id for what the set it. The great thing about the service factory is that we can make scoped services onto the controller database without the facade having to know what the model id is.

@barrettj12 just did some good work with a similar problem for agent version. See https://github.com/juju/juju/blob/main/domain/modelagent/service/service.go for example of how I believe this should be solved.

You then removed the need for having to know the model id inside the facade. It also keeps the foot print of what model config should know about small.

Reply:
Does the model config service need a validator to make sure that the secret backend can't be set via model config anymore?

Reply:
Yes, `SecretBackendChecker` now prevents this happen.